### PR TITLE
[BACKLOG-21825] - Added unit tests for modelAdapter._convertFilterToI…

### PR DESCRIPTION
**Pull request by graimundo**
…nternal

- Fixed bugs for edge cases and made conversion less lenient for filters not explicitly considered.

### CODE REVIEW

**Comment by graimundo** [comment]
@pentaho/millenniumfalcon please review

File: impl/client/src/main/javascript/web/pentaho/visual/base/modelAdapter.js [hunk 1]
```diff
@@ -589,77 +604,102 @@ define([
...
+                  __context
+                );
+              }, this));
             }
-          });
 
-          if(equalsMap === null) {
-            return filter;
-          }
+            return new AndFilter({operands: operands});
```
Comment (nantunes) [inline impl/client/src/main/javascript/web/pentaho/visual/base/modelAdapter.js]: What happens if `equalsMap` is empty? Is an empty `and` equivalent to `true`, like in `isEqual` case bellow?
